client/client_test.go

```diff
@@ -918,6 +936,81 @@ func testListTarget(t *testing.T, rootType string) {
 	assert.Equal(t, currentTarget, newCurrentTarget, "current target does not match")
 }
 
+func testListTargetWithDelegates(t *testing.T, rootType string) {
+	// Temporary directory where test files will be created
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	defer os.RemoveAll(tempBaseDir)
+
+	assert.NoError(t, err, "failed to create a temporary directory: %s", err)
+
+	gun := "docker.com/notary"
+
+	ts, mux, keys := simpleTestServer(t)
+	defer ts.Close()
+
+	repo, _ := initializeRepo(t, rootType, tempBaseDir, gun, ts.URL, false)
+
+	// tests need to manually boostrap timestamp as client doesn't generate it
+	err = repo.tufRepo.InitTimestamp()
+	assert.NoError(t, err, "error creating repository: %s", err)
+
+	latestTarget := addTarget(t, repo, "latest", "../fixtures/intermediate-ca.crt")
+	addTarget(t, repo, "current", "../fixtures/intermediate-ca.crt")
+
+	// setup delegated targets/level1 role
+	k, err := repo.CryptoService.Create("targets/level1", rootType)
+	assert.NoError(t, err)
+	r, err := data.NewRole("targets/level1", 1, []string{k.ID()}, nil, nil)
+	assert.NoError(t, err)
+	repo.tufRepo.UpdateDelegations(r, []data.PublicKey{k})
+	delegatedTarget := addTarget(t, repo, "current", "../fixtures/root-ca.crt", "targets/level1")
+	otherTarget := addTarget(t, repo, "other", "../fixtures/root-ca.crt", "targets/level1")
+
+	// Apply the changelist. Normally, this would be done by Publish
+
+	// load the changelist for this repo
+	cl, err := changelist.NewFileChangelist(
+		filepath.Join(tempBaseDir, "tuf", filepath.FromSlash(gun), "changelist"))
+	assert.NoError(t, err, "could not open changelist")
+
+	// apply the changelist to the repo
+	err = applyChangelist(repo.tufRepo, cl)
+	assert.NoError(t, err, "could not apply changelist")
+
+	_, ok := repo.tufRepo.Targets["targets/level1"].Signed.Targets["current"]
+	assert.True(t, ok)
+	_, ok = repo.tufRepo.Targets["targets/level1"].Signed.Targets["other"]
+	assert.True(t, ok)
+
+	fakeServerData(t, repo, mux, keys)
+
+	targets, err := repo.ListTargets(data.CanonicalTargetsRole, "targets/level1")
+	assert.NoError(t, err)
+
+	// Should be two targets
```

Comment:
"three targets"  :)  Although the assert is pretty clear, so this comment may just be superfluous.